source/extensions/filters/http/stateful_session/stateful_session.h

```diff
@@ -58,19 +78,63 @@ using PerRouteStatefulSessionConfigSharedPtr = std::shared_ptr<PerRouteStatefulS
 class StatefulSession : public Http::PassThroughFilter,
                         public Logger::Loggable<Logger::Id::filter> {
 public:
-  StatefulSession(StatefulSessionConfigSharedPtr config) : config_(std::move(config)) {}
+  StatefulSession(StatefulSessionConfigSharedPtr config, Stats::Scope& scope,
+                  const std::string& hcm_prefix)
+      : config_(std::move(config)), scope_(scope), hcm_prefix_(hcm_prefix),
+        stats_(generateStats(hcm_prefix_, config_->statPrefixOverride(), scope_)) {}
```

Comment:
I think the stats should be part of the `StatefulSessionConfig` rather then part of the filter instance to avoid we construct it from the scope again and again. It's inefficient.